### PR TITLE
add simultaneous support 10m and 30m RTC_GAMMA/WATER_MAP jobs

### PR DESCRIPTION
**Pull request by asjohnston-asf**
This makes 10m RTC_GAMMA available in these environments where 30m was already available:
- hyp3-edc-prod
- hyp3-edc-uat
- hyp3-enterprise-test
- hyp3-watermap
- hyp3-cargill

This makes 10m WATER_MAP available in these environments where 30m was already available:
- hyp3-enterprise-test
- hyp3-watermap

This makes 30m RTC_GAMMA and WATER_MAP more cost effective in the `hyp3-streamflow` deployment where both 30m and 10m were already available.

### CODE REVIEW

**Comment by jtherrmann** [comment]
Confirmed that `instance_types` is set to `r6id.xlarge,r6id.2xlarge,r6id.4xlarge,r6id.8xlarge,r5dn.xlarge,r5dn.2xlarge,r5d.xlarge,r5d.2xlarge` for all the relevant deployments.

File: apps/step-function.json.j2 [hunk 1]
```diff
@@ -36,20 +36,61 @@
...
+          ],
           "Next": "USE_LANDSAT_MEMORY"
+        },
+        {
+          "And": [
+            {
+              "Or": [
+                {
+                  "Variable": "$.job_type",
+                  "StringEquals": "RTC_GAMA"
```
Comment (jtherrmann) [inline apps/step-function.json.j2:76]:
```suggestion
                  "StringEquals": "RTC_GAMMA"
```

File: job_spec/RIVER_WIDTH.yml [hunk 2]
```diff
@@ -32,6 +32,8 @@ RIVER_WIDTH:
       command:
         - ++process
         - water_map
+        - ++omp-num-threads
```
Comment (jtherrmann) [inline job_spec/RIVER_WIDTH.yml:35]:
```suggestion
        # TODO determine final value
        - ++omp-num-threads
```

**Comment by jtherrmann** [comment]
Looks good! :+1:

**Review by jhkennedy — APPROVED** [review]
Looks good to me!

**Pull request merged**